client_test.go

```diff
@@ -43,37 +44,67 @@ func TestDecodeResponse(t *testing.T) {
 	testCases := []struct {
 		name     string
 		value    interface{}
+		expected interface{}
 		body     io.Reader
 		hasError bool
 	}{
 		{
-			name:  "nil input",
-			value: nil,
-			body:  bytes.NewReader([]byte("")),
+			name:     "nil input",
+			value:    nil,
+			body:     bytes.NewReader([]byte("")),
+			expected: nil,
 		},
 		{
-			name:  "string input",
-			value: &stringInput,
-			body:  bytes.NewReader([]byte("test")),
+			name:     "string input",
+			value:    &stringInput,
+			body:     bytes.NewReader([]byte("test")),
+			expected: "test",
 		},
 		{
 			name:  "map input",
 			value: &map[string]interface{}{},
 			body:  bytes.NewReader([]byte(`{"test": "test"}`)),
+			expected: map[string]interface{}{
+				"test": "test",
+			},
 		},
 		{
 			name:     "reader return error",
 			value:    &stringInput,
 			body:     &errorReader{err: errors.New("dummy")},
 			hasError: true,
 		},
+		{
+			name:  "audio text input",
+			value: &audioTextResponse{},
+			body:  bytes.NewReader([]byte("test")),
+			expected: audioTextResponse{
+				Text: "test",
+			},
+		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			err := decodeResponse(tc.body, tc.value)
-			if (err != nil) != tc.hasError {
-				t.Errorf("Unexpected error: %v", err)
+			if tc.hasError {
+				if err == nil {
+					t.Error("Unexpected nil error")
+				}
+			} else {
```

Comment:
we can reduce the amount of `else` by using early return pattern:

```
if something {
// code
    return
}

// else block
```

Reply:
Of course! It would be better to use github.com/stretchr/testify/assert for testing, as it eliminates the need for if/else statements for assertions. However, I didn't use it because it doesn't have any external dependencies. Would you like to use it or keep it dependency-free?

Reply:
@WqyJh staying dependency free is important for the project, so let's not use the library. We have an internal lib of helpers though: https://github.com/sashabaranov/go-openai/blob/master/internal/test/checks/checks.go